### PR TITLE
Lower bcrypt cost when testing

### PR DESCRIPTION
**Pull request by bl-nero**
This PR should minimize the number of timeouts from the flaky test detector if the affected tests contain calls to `bcrypt` using the `IdentityService`.

This PR is similar to https://github.com/gravitational/teleport.e/pull/3551.

### CODE REVIEW

**Review by espadolini — APPROVED** [review]

File: lib/cache/cache_test.go [hunk 1]
```diff
@@ -245,14 +245,14 @@ func newPackWithoutCache(dir string, opts ...packOption) (*testPack, error) {
 	p.provisionerS = local.NewProvisioningService(p.backend)
 	p.eventsS = newProxyEvents(local.NewEventsService(p.backend), cfg.ignoreKinds)
 	p.presenceS = local.NewPresenceService(p.backend)
-	p.usersS = local.NewIdentityService(p.backend)
+	p.usersS = local.NewTestIdentityService(p.backend)
```
Comment (codingllama) [inline lib/cache/cache_test.go:248]: Suggestion: Extract `local.NewTestIdentityService(p.backend)` and use it throughout?

File: lib/services/local/users.go [hunk 2]
```diff
@@ -62,7 +63,8 @@ var GlobalSessionDataMaxEntries = 5000 // arbitrary
 // user accounts as well
 type IdentityService struct {
 	backend.Backend
-	log logrus.FieldLogger
+	log        logrus.FieldLogger
+	bcryptCost int
```
Comment (codingllama) [inline lib/services/local/users.go:67]: Suggestion: Explicitly set the default cost on NewIdentityService?

I realize bcrypt will default `0` to bcrypt.DefaultCost, but I think being more explicit is likely to cause less confusion.

**Review by codingllama — COMMENTED** [review]
Thanks for doing this!

File: lib/services/local/users.go (hunk 2 above)
Comment (bl-nero) [reply lib/services/local/users.go:67]: Good point.

**Review by codingllama — APPROVED** [review]

**Pull request merged**